addons/im_livechat/static/tests/embed/feedback_panel_tests.js

```diff
@@ -0,0 +1,108 @@
+    await startServer();
+    await loadDefaultConfig();
+    const { root } = await start();
+    await click(".o-livechat-LivechatButton");
+    assert.containsOnce(root, ".o-mail-ChatWindow");
+    await click(".o-mail-ChatWindow-command[title*='Close']");
+    assert.containsNone(root, ".o-livechat-LivechatButton");
+});
+
+QUnit.test("Close without feedback", async (assert) => {
+    await startServer();
+    await loadDefaultConfig();
+    const { root } = await start({
+        mockRPC(route) {
+            if (route === "/im_livechat/visitor_leave_session") {
+                assert.step(route);
+            }
+            if (route === "/im_livechat/feedback") {
+                assert.step(route);
+            }
+        },
+    });
+    await click(".o-livechat-LivechatButton");
+    assert.containsOnce(root, ".o-mail-ChatWindow");
+    await insertText(".o-mail-Composer-input", "Hello World!");
+    await afterNextRender(() => triggerHotkey("Enter"));
+    await click(".o-mail-ChatWindow-command[title*='Close']");
+    await click("button:contains('Close')");
+    assert.containsNone(root, ".o-livechat-LivechatButton");
+    assert.verifySteps(["/im_livechat/visitor_leave_session"]);
+});
+
+QUnit.test("Feedback with rating and comment", async (assert) => {
+    await startServer();
+    await loadDefaultConfig();
+    let messageCount = 0;
+    const { root } = await start({
+        mockRPC(route, args) {
+            if (route === "/im_livechat/visitor_leave_session") {
+                assert.step(route);
+            }
+            if (route === "/im_livechat/feedback") {
+                assert.step(route);
+            }
+            if (route === "/im_livechat/chat_post") {
+                if (messageCount === 1) {
+                    assert.strictEqual(
+                        args.message_content,
+                        `Rating: ${RATING_TO_EMOJI[RATING.GOOD]}`
+                    );
+                } else if (messageCount === 2) {
+                    assert.strictEqual(args.message_content, "Rating reason: Good job!");
+                }
+                messageCount++;
+                assert.step(route);
+            }
+        },
+    });
+    await click(".o-livechat-LivechatButton");
+    assert.containsOnce(root, ".o-mail-ChatWindow");
+    await insertText(".o-mail-Composer-input", "Hello World!");
+    await afterNextRender(() => triggerHotkey("Enter"));
+    await click(".o-mail-ChatWindow-command[title*='Close']");
+    await click(`img[data-alt="${RATING.GOOD}"]`);
+    await insertText("textarea[placeholder='Explain your note']", "Good job!");
+    await click("button:contains(Send)");
+    assert.containsOnce(root, "p:contains(Thank you for your feedback)");
+    assert.verifySteps([
+        "/im_livechat/chat_post",
+        "/im_livechat/visitor_leave_session",
+        "/im_livechat/feedback",
+        "/im_livechat/chat_post",
+        "/im_livechat/chat_post",
+    ]);
```

Comment:
It's best to verifySteps as early as you can, so we understand clearly which action triggers a step.
Seeing `chat_post` 3 times look off when it's grouped together

Reply:
Done. Feedback should be refactored in order to only call the feedback route. Then the route should post the two other messages. It was already done like this before so I think we can improve this in a follow up task :)